ports/mimxrt/machine_uart.c

```diff
@@ -81,6 +81,7 @@ static const iomux_table_t iomux_table_uart[] = {
 static const iomux_table_t iomux_table_uart_cts_rts[] = {
     IOMUX_TABLE_UART_CTS_RTS
 };
+static bool uart_active[MICROPY_HW_UART_NUM];
```

Comment:
Instead of a variable like this, is it possible to use a LPUART register to determine if it's enabled or not?  Eg `self->lpuart->CTRL != 0` (because at least one of the TE or RE bits must be enabled for the UART to be active).

Reply:
There is a problem with the 1176, in that access to the CTRL register for testing already causes the exception. Using local flags is reliable.

Reply:
But why does the 1176 have the issue, is it because the peripheral clock is not yet enabled?  Surely there's a bit somewhere that can be used to tell if the UART can safely be disabled.